data/collector.go

```diff
@@ -28,6 +29,14 @@ var sleepCaptureCutoff = 2 * time.Millisecond
 // CaptureFunc allows the creation of simple Capturers with anonymous functions.
 type CaptureFunc func(ctx context.Context, params map[string]*anypb.Any) (interface{}, error)
 
+type contextKey string
```

Comment:
Is this type used anywhere besides below? Wondering why we can't just do `const FromDMContextKey string = "fromDataManagement"`

Reply:
Ah I'm seeing this now. According to [the docs](https://pkg.go.dev/context#WithValue), seems like context keys often have concrete type struct{}. Should we try that?

Reply:
Sure, changed - since `FromDMContextKey` is now of type struct{} and not convertible to string, I added a separate string constant to check the Extra struct with.